### PR TITLE
[Bug]: Fix `pimcore:build:classes` requiring DB and Cache

### LINKED ISSUE #15520: [Bug]: pimcore:build:classes requires DB (and other services) for multiple class definitions
### Pimcore version

10.6

### Steps to reproduce

* Shutdown database (or execute the following command during build-time)
* Run `bin/console pimcore:build:classes`

### Expected / actual behavior

* Actual behavior:
  * running `bin/console pimcore:build:classes` results in an error:
    ```
    Next Doctrine\DBAL\Exception\ConnectionException: An exception occurred in driver: SQLSTATE[HY000] [2002] Connection refused in /var/www/pimcore/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:112
    Stack trace:
    #0 /var/www/pimcore/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php(182): Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\DBAL\Driver\PDO\Exception))
    #1 /var/www/pimcore/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php(169): Doctrine\DBAL\DBALException::wrapException(Object(Doctrine\DBAL\Driver\PDO\MySQL\Driver), Object(Doctrine\DBAL\Driver\PDO\Exception), 'An exception oc...')
    #2 /var/www/pimcore/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOMySql/Driver.php(31): Doctrine\DBAL\DBALException::driverException(Object(Doctrine\DBAL\Driver\PDO\MySQL\Driver), Object(Doctrine\DBAL\Driver\PDO\Exception))
    #3 /var/www/pimcore/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(412): Doctrine\DBAL\Driver\PDOMySql\Driver->connect(Array, '', '', Array)
    #4 /var/www/pimcore/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(1952): Doctrine\DBAL\Connection->connect()
    #5 /var/www/pimcore/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(1290): Doctrine\DBAL\Connection->getWrappedConnection()
    #6 /var/www/pimcore/vendor/pimcore/pimcore/lib/Db/PimcoreExtensionsTrait.php(92): Doctrine\DBAL\Connection->executeQuery('SELECT id FROM ...', Array, Array, NULL)
    #7 /var/www/pimcore/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(1119): Pimcore\Db\Connection->executeQuery('SELECT id FROM ...', Array, Array)
    #8 /var/www/pimcore/vendor/pimcore/pimcore/models/Tool/Targeting/TargetGroup/Listing/Dao.php(33): Doctrine\DBAL\Connection->fetchFirstColumn('SELECT id FROM ...', Array)
    #9 [internal function]: Pimcore\Model\Tool\Targeting\TargetGroup\Listing\Dao->load()
    #10 /var/www/pimcore/vendor/pimcore/pimcore/lib/Model/AbstractModel.php(246): call_user_func_array(Array, Array)
    #11 /var/www/pimcore/vendor/pimcore/pimcore/models/DataObject/ClassDefinition/Data/TargetGroup.php(87): Pimcore\Model\AbstractModel->__call('load', Array)
    #12 /var/www/pimcore/vendor/pimcore/pimcore/models/DataObject/ClassDefinition/Data/TargetGroup.php(128): Pimcore\Model\DataObject\ClassDefinition\Data\TargetGroup->configureOptions()
    #13 /var/www/pimcore/var/classes/definition_CustomerSegment.php(255): Pimcore\Model\DataObject\ClassDefinition\Data\TargetGroup::__set_state(Array)
    #14 /var/www/pimcore/vendor/pimcore/pimcore/bundles/CoreBundle/Command/ClassesDefinitionsBuildCommand.php(63): include('/var/www/pimcor...')
    #15 /var/www/pimcore/vendor/symfony/console/Command/Command.php(298): Pimcore\Bundle\CoreBundle\Command\ClassesDefinitionsBuildCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
    #16 /var/www/pimcore/vendor/symfony/console/Application.php(1058): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
    #17 /var/www/pimcore/vendor/symfony/framework-bundle/Console/Application.php(96): Symfony\Component\Console\Application->doRunCommand(Object(Pimcore\Bundle\CoreBundle\Command\ClassesDefinitionsBuildCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
    #18 /var/www/pimcore/vendor/symfony/console/Application.php(301): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand(Object(Pimcore\Bundle\CoreBundle\Command\ClassesDefinitionsBuildCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
    #19 /var/www/pimcore/vendor/symfony/framework-bundle/Console/Application.php(82): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
    #20 /var/www/pimcore/vendor/symfony/console/Application.php(171): Symfony\Bundle\FrameworkBundle\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
    #21 /var/www/pimcore/bin/console(48): Symfony\Component\Console\Application->run()
    #22 {main}
    ```
  * Affected after a quick scan (probably even more ClassDefinitions affected)
    * `models/DataObject/ClassDefinition/Data/TargetGroup.php`
    * `models/DataObject/ClassDefinition/Data/QuantityValue.php`

* Expected behavior:
  * building php classes should not rely on any external services (db, cache) etc.
  * `bin/console pimcore:build:classes` should be executable during build-time

### PR DESCRIPTION
**Pull request by kingjia90**
…options when on cli

  
  

## Changes in this pull request  
Resolves https://github.com/pimcore/pimcore/issues/15520

## Additional info
Following https://github.com/pimcore/pimcore/pull/13752 approach and what's on `models/DataObject/ClassDefinition/Data/User.php`

### <samp>🤖 Generated by Copilot at a10f63d</samp>

This pull request enhances the performance and reliability of several data object field types and the class definitions build command. It reduces unnecessary database queries, optimizes the initialization of options, and prevents cache issues by using serialization hooks and disabling the cache temporarily. It affects the files `ClassesDefinitionsBuildCommand.php`, `QuantityValue.php`, `TargetGroup.php`, and `TargetGroupMultiselect.php`.

### <samp>🤖 Generated by Copilot at a10f63d</samp>

> _We are the masters of the cache_
> _We disable and restore with a flash_
> _We optimize the queries and the fields_
> _We unleash the power of the class definitions_

### <samp>🤖 Generated by Copilot at a10f63d</samp>

*  Disable and re-enable cache for class definitions build command ([link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-26ffe385bbb10aac34e10682fddb7958ce4ebb4375a578b885f653ac0c47334fR55-R56),[link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-26ffe385bbb10aac34e10682fddb7958ce4ebb4375a578b885f653ac0c47334fR84-R86))
*  Add `__wakeup` and `init` methods to `QuantityValue`, `TargetGroup`, and `TargetGroupMultiselect` classes to initialize options from database when unserialized ([link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-d2564b1d2701cc3ab300240f945b53cb21a2466232ad24ad2e5d75f10f5789faL172-R192),[link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-29d45b2bf798808223d848c490cedf99866bcfd2a11e3a2801a1495ce9bd94adL33-R53),[link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-3814f98239105bcbfdf441a881bfdf74bf58acf84ad6178813f4e5566caeb71eL33-R55))
*  Call `init` methods in `getDataForResource` methods of `QuantityValue`, `TargetGroup`, and `TargetGroupMultiselect` classes to ensure options are initialized before returning data ([link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-d2564b1d2701cc3ab300240f945b53cb21a2466232ad24ad2e5d75f10f5789faR528-R529),[link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-29d45b2bf798808223d848c490cedf99866bcfd2a11e3a2801a1495ce9bd94adR86),[link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-3814f98239105bcbfdf441a881bfdf74bf58acf84ad6178813f4e5566caeb71eR92-R100))
*  Modify `__set_state` methods of `QuantityValue`, `TargetGroup`, and `TargetGroupMultiselect` classes to only call `configureOptions` methods if in admin mode, to avoid unnecessary database queries when cached ([link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-d2564b1d2701cc3ab300240f945b53cb21a2466232ad24ad2e5d75f10f5789faL830-R854),[link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-29d45b2bf798808223d848c490cedf99866bcfd2a11e3a2801a1495ce9bd94adL126-R147),[link](https://github.com/pimcore/pimcore/pull/15726/files?diff=unified&w=0#diff-3814f98239105bcbfdf441a881bfdf74bf58acf84ad6178813f4e5566caeb71eL64-R84))

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/QuantityValue.php [hunk 1]
```diff
@@ -169,6 +169,26 @@ class QuantityValue extends Data implements ResourcePersistenceAwareInterface, Q
...
 
+    public function __wakeup()
+    {
+        $this->init();
+    }
+
+    /**
+     * @internal
+     *
+     * @return TargetGroup
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/QuantityValue.php]:
```suggestion
     * @return $this
```

File: models/DataObject/ClassDefinition/Data/TargetGroup.php [hunk 2]
```diff
@@ -30,6 +30,26 @@ class TargetGroup extends Model\DataObject\ClassDefinition\Data\Select
...
 
+    public function __wakeup()
+    {
+        $this->init();
+    }
+
+    /**
+     * @internal
+     *
+     * @return TargetGroup
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/TargetGroup.php]:
```suggestion
     * @return $this
```

File: models/DataObject/ClassDefinition/Data/TargetGroupMultiselect.php [hunk 3]
```diff
@@ -30,6 +30,26 @@ class TargetGroupMultiselect extends Model\DataObject\ClassDefinition\Data\Multi
...
 
+    public function __wakeup()
+    {
+        $this->init();
+    }
+
+    /**
+     * @internal
+     *
+     * @return TargetGroup
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/TargetGroupMultiselect.php]:
```suggestion
     * @return $this
```

File: models/DataObject/ClassDefinition/Data/QuantityValue.php [hunk 4]
```diff
@@ -169,6 +169,26 @@ class QuantityValue extends Data implements ResourcePersistenceAwareInterface, Q
         'unit' => 'varchar(64)',
     ];
 
+    public function __wakeup()
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/QuantityValue.php]:
```suggestion
    public function __wakeup(): void
```

File: models/DataObject/ClassDefinition/Data/TargetGroup.php [hunk 5]
```diff
@@ -30,6 +30,26 @@ class TargetGroup extends Model\DataObject\ClassDefinition\Data\Select
      */
     public $fieldtype = 'targetGroup';
 
+    public function __wakeup()
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/TargetGroup.php]:
```suggestion
    public function __wakeup(): void
```

File: models/DataObject/ClassDefinition/Data/TargetGroupMultiselect.php [hunk 6]
```diff
@@ -30,6 +30,26 @@ class TargetGroupMultiselect extends Model\DataObject\ClassDefinition\Data\Multi
      */
     public $fieldtype = 'targetGroupMultiselect';
 
+    public function __wakeup()
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/TargetGroupMultiselect.php]:
```suggestion
    public function __wakeup(): void
```

File: models/DataObject/ClassDefinition/Data/QuantityValue.php [hunk 7]
```diff
@@ -169,6 +169,26 @@ class QuantityValue extends Data implements ResourcePersistenceAwareInterface, Q
...
+    {
+        $this->init();
+    }
+
+    /**
+     * @internal
+     *
+     * @return $this
+     */
+    protected function init(): static
```
Comment (robertSt7) [inline models/DataObject/ClassDefinition/Data/QuantityValue.php]: I am not sure why all the `init` functions needs to be protected instead of private?

**Review by robertSt7 — APPROVED** [review]

**Pull request merged**